sos/report/plugins/vdo.py

```diff
@@ -28,6 +28,32 @@ def setup(self):
         vdos = self.collect_cmd_output('vdo list --all')
         for vdo in vdos['output'].splitlines():
             self.add_cmd_output(f"vdo status -n {vdo}")
-        self.add_cmd_output('vdostats --human-readable')
+        self.add_cmd_output([
+            'vdostats --human-readable',
+            'vdostats --verbose',
+        ])
+        vdo_cols1 = ('vdo_slab_size,vdo_header_size,vdo_minimum_io_size,'
+                     'vdo_block_map_cache_size,vdo_block_map_era_length,'
+                     'vdo_write_policy,vdo_max_discard')
+
+        vdo_cols2 = ('vdo_ack_threads,vdo_bio_rotation,vdo_bio_threads,'
+                     'vdo_cpu_threads,vdo_hash_zone_threads,'
+                     'vdo_logical_threads,vdo_physical_threads')
+        vdo_cols3 = ('vdo_compression,vdo_deduplication,'
+                     'vdo_use_metadata_hints,vdo_use_sparse_index,'
+                     'vdo_index_state,vdo_index_memory_size')
+        self.add_cmd_output([f"lvs -a -o +{cols}"
+                             for cols in [vdo_cols1, vdo_cols2]])
+        lvm_vdos = self.collect_cmd_output(f"lvs -a -o +{vdo_cols3}")
+        if lvm_vdos['status'] == 0:
+            for vdo in lvm_vdos['output'].splitlines():
+                # we can find the pool and pool data maps in the output
+                # of lvs, in the column Volume type,
+                # marked as either 'd' or 'D'
+                if vdo.split()[2].startswith("D"):
+                    vdo_path = f"{vdo.split()[1]}-{vdo.split()[0].strip('[]')}"
```

Comment:
Are we sure `vdo` will always contain at least 3 words..? (we do rely on such assumptions, sometimes. I dont know `vdo` at all to assess if we can safely rely here).

As we call `vdo.split()` three times in a row, isn't it faster and also better to understand something like:

     foo, bar, volume = vdo.split()[:3]

but with some more appropriate names?

Reply:
The output of lvs should always be the same, because we rely on standard columns in the output, and we add the rest with the '-o+'. If the default column order change then we'll have to revisit other plugins as well. The three columns I rely on are:

`  LV             VG    Attr       `

But the device in /dev/ has the format VG-LV, and the VG is always surrounded (in this case) by [], so I have to split and then strip. I'll try to rewrite vdo_path using your approach.